netbox_onboarding/netdev_keeper.py

```diff
@@ -257,20 +260,21 @@ def get_onboarding_facts(self):
 
             napalm_device.open()
 
-            logging.info("COLLECT: device facts")
+            logger.info("COLLECT: device facts")
             self.facts = napalm_device.get_facts()
 
-            logging.info("COLLECT: device interface IPs")
+            logger.info("COLLECT: device interface IPs")
             self.ip_ifs = napalm_device.get_interfaces_ip()
 
             try:
                 module_name = PLUGIN_SETTINGS["onboarding_extensions_map"].get(self.napalm_driver)
-                module = importlib.import_module(module_name)
-                driver_addon_class = module.OnboardingDriverExtensions(napalm_device=napalm_device)
-                self.onboarding_class = driver_addon_class.onboarding_class
-                self.driver_addon_result = driver_addon_class.ext_result
+                if module_name:
```

Comment:
Why was the function failing without this check? We already have a `try...except` wrapper around this block - are we not catching the right exception class(es)?

We need some test automation around the `get_onboarding_facts()` function, clearly - would you be able and willing to take a pass at adding some?

Reply:
The try except block would raise OnboardException at line 281 with a `fail-general` reason and a message of `'NoneType' object has no attribute 'startswith'`. The problem, as I understand, is that if there is no driver addon class defined, there is no need to execute lines 268 - 271. They fail during normal execution because they're trying to pull attributes from a NoneType object, which raises an AttributeError. This is caught by the except block which is grabbing all Exceptions on line 281.

Reply:
Gotcha. So the code was raising an `AttributeError` but line 276 was only expecting an `ImportError`, hence the failure. This fix makes sense to me then.

Reply:
I'll take a stab at adding some tests for the `get_onboarding_facts()` function as well :).